testing_and_setup/compass/ocean/global_ocean/ARM60to10/init/define_base_mesh.py

```diff
@@ -1,41 +1,131 @@
+    plt.colorbar(im, shrink=.9)
+    plt.title(varName)
 
 
 def cellWidthVsLatLon():
-    lat = np.arange(-90, 90.01, 0.1)
-    lon = np.arange(-180, 180.01, 0.1)
+    """
+    Create cell width array for this mesh on a regular latitude-longitude grid.
+    Returns
+    -------
+       cellWidth : numpy.ndarray
+            m x n array, entries are desired cell width in km
+       lat : numpy.ndarray
+            latitude, vector of length m, with entries between -90 and 90,
+            degrees
+       lon : numpy.ndarray
+            longitude, vector of length n, with entries between -180 and 180,
+            degrees
+    """
+    # To speed up for testing, set following line to 1.0 degrees
+    dlon = 1.0
+    dlat = dlon
+    nlon = int(360. / dlon) + 1
+    nlat = int(180. / dlat) + 1
+    lon = np.linspace(-180., 180., nlon)
+    lat = np.linspace(-90., 90., nlat)
 
+    # Create cell width vs latitude for Atlantic and Pacific basins
     QU1 = np.ones(lat.size)
     EC60to30 = mdt.EC_CellWidthVsLat(lat)
     RRS30to10 = mdt.RRS_CellWidthVsLat(lat, 30, 10)
-
     AtlNH = RRS30to10
-    AtlGrid = mdt.mergeCellWidthVsLat(lat, EC60to30, AtlNH, 0, 6)
-
+    AtlVsLat = mdt.mergeCellWidthVsLat(lat, EC60to30, AtlNH, 0, 6)
     PacNH = mdt.mergeCellWidthVsLat(lat, 30 * QU1, RRS30to10, 50, 10)
-    PacGrid = mdt.mergeCellWidthVsLat(lat, EC60to30, PacNH, 0, 6)
+    PacVsLat = mdt.mergeCellWidthVsLat(lat, EC60to30, PacNH, 0, 6)
+
+    # Expand from 1D to 2D
+    _, AtlGrid = np.meshgrid(lon, AtlVsLat)
+    _, PacGrid = np.meshgrid(lon, PacVsLat)
+
+    # Signed distance of Atlantic region
+    fc = read_feature_collection('Atlantic_region.geojson')
+    signedDistance = signed_distance_from_geojson(fc, lon, lat,
+                                                 max_length=0.25)
+
+    # Merge Atlantic and Pacific distrubutions smoothly
+    transitionWidth = 500.0e3  # [m]
+    maskSmooth = 0.5 * (1 + np.tanh(signedDistance / transitionWidth))
+    cellWidthSmooth = PacGrid * maskSmooth + AtlGrid * (1 - maskSmooth)
 
-    cellWidth = mdt.AtlanticPacificGrid(lat, lon, AtlGrid, PacGrid)
+    # Merge Atlantic and Pacific distrubutions with step function
+    maskStep = 0.5 * (1 + np.sign(signedDistance))
+    cellWidthStep = PacGrid * maskStep + AtlGrid * (1 - maskStep)
 
-    import matplotlib.pyplot as plt
-    import matplotlib
-    matplotlib.use('Agg')
+    # Create a land mask that is 1 over land
+    fc = read_feature_collection('Americas_land_mask.geojson')
+    Americas_land_mask = mask_from_geojson(fc, lon, lat)
+    fc = read_feature_collection('Europe_Africa_land_mask.geojson')
+    Europe_Africa_land_mask = mask_from_geojson(fc, lon, lat)
+    landMask = np.fmax(Americas_land_mask, Europe_Africa_land_mask)
+
+    # Merge: step transition over land, smooth transition over water
+    cellWidth = cellWidthStep * landMask + cellWidthSmooth * (1 - landMask)
+
+    # save signed distance to a file
+    # da = xarray.DataArray(signedDistance,
```

Comment:
@xylar this is why we were confused on the phone. In the SO60to10 you had these lines to save signed distance, not cell width. Is signed distance useful in a netcdf file, since I'm making a plot of it below? I think I can delete these lines.

Reply:
No, that was just for debugging.  But I definitely do want the cell widths saved to a NetCDF file but I think the code that calls `define_base_mesh.py` actually does the saving so you don't need to do it.